Sources/TensorFlow/Core/Tensor.swift

```diff
@@ -35,6 +35,9 @@ public struct Tensor<Scalar: TensorFlowScalar> {
   /// - Note: `handle` is public to allow user defined ops, but should not normally be used.
   public let handle: TensorHandle<Scalar>
 
+  /// An internal marker to identify scalar zero tensors, for use in optimizations.
+  public var _isScalarZero = false
```

Comment:
I believe you can avoid publicly exposing `_isScalarZero` by marking it as `@usableFromInline`:

```suggestion
  @usableFromInline
  internal var _isScalarZero = false
```

Reply:
Great, thanks. That's definitely preferred over having this implementation detail exposed publicly.